Extension/src/LanguageServer/Providers/renameProvider.ts

```diff
@@ -24,6 +24,16 @@ export class RenameProvider implements vscode.RenameProvider {
     }
 
     public async provideRenameEdits(document: vscode.TextDocument, position: vscode.Position, newName: string, _token: vscode.CancellationToken): Promise<vscode.WorkspaceEdit | undefined> {
+        // Bypass the normal rename processing during Extract to function,
```

Comment:
I've expressed my concerns about leveraging "Rename" for this feature at all.  It doesn't seem to be the right approach, regardless of whether TypeScript uses it (which seems to have been a convenience).  Have you followed up with the VS Code team with an issue to address this with UI that doesn't need to be awkwardly routed through Rename?

Reply:
I haven't filed a VS Code issue because they're unlikely to change the TypeScript refactor experience -- you're free to file an issue if you want. I think the plan is just to change this to prompt for the name before the extracted code is generated, but in a follow up change for 1.18.1

Reply:
See the TODO in handleExtractToFunction:
```ts
        // TODO: Show a quick pick to get the name before generating the code.
        // That would allow the formatting to be done without waiting for the rename.
        // Also, it's less error prone and eliminates a class of bugs in which the
        // rename position can be incorrect.
```